### PR TITLE
chore: align Docker with upstream Docker infra

### PR DESCRIPTION
**Pull request by mxschmitt**
This aligns the build infra with the upstream infra which we use now across all the language bindings.

As a drive by, we get:

- arm images
- pwuser support
- easier maintenance

### CODE REVIEW

File: utils/docker/Dockerfile.focal [hunk 1]
```diff
@@ -2,26 +2,29 @@ FROM ubuntu:focal
...
+    apt-get install -y --no-install-recommends \
+    openjdk-11-jdk maven \
+    curl unzip && \
+    rm -rf /var/lib/apt/lists/* && \
+    # Create the pwuser
+    adduser pwuser  
 
 ENV JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64
 
-# Install utilities required for downloading driver
```
Comment (yury-s) [inline utils/docker/Dockerfile.focal:10]: Is there any reason to change this? Let's keep the comment at least.

**Review by yury-s — COMMENTED** [review]
If we use upstream scripts let's add README with instructions on how to keep them up to date including links to the upstream code.

File: utils/docker/publish_docker.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  echo "ERROR: GITHUB_SHA env variable must be specified"
+  exit 1
+fi
+
+BIONIC_TAGS=(
+  "next-bionic"
+  "v${PW_VERSION}-bionic"
+)
+if [[ "$RELEASE_CHANNEL" == "stable" ]]; then
+  BIONIC_TAGS+=("bionic")
```
Comment (yury-s) [inline utils/docker/publish_docker.sh:39]: We don't publish binonic for java.

File: utils/docker/publish_docker.sh [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  for ((i = 0; i < ${#TAGS[@]}; i++)) do
+    local TAG="${TAGS[$i]}"
+    tag_and_push "${MCR_IMAGE_NAME}:localbuild" "playwright.azurecr.io/public/${MCR_IMAGE_NAME}:${TAG}-${ARCH}"
+  done
+}
+
+publish_docker_manifest () {
+  local FLAVOR="$1"
+  local TAGS=()
+  if [[ "$FLAVOR" == "bionic" ]]; then
```
Comment (yury-s) [inline utils/docker/publish_docker.sh:92]: no bionic for java

File: .github/workflows/publish_docker_release.yml [hunk 4]
```diff
@@ -17,20 +17,12 @@ jobs:
...
-        then
-          echo "Wrong TAG_NAME format: $TAG_NAME"
-          exit 1
-        fi
-        ./scripts/tag_image_and_push.sh playwright-java:localbuild-focal playwright.azurecr.io/public/playwright/java:latest
-        ./scripts/tag_image_and_push.sh playwright-java:localbuild-focal playwright.azurecr.io/public/playwright/java:focal
-        ./scripts/tag_image_and_push.sh playwright-java:localbuild-focal playwright.azurecr.io/public/playwright/java:${TAG_NAME}
-        ./scripts/tag_image_and_push.sh playwright-java:localbuild-focal playwright.azurecr.io/public/playwright/java:${TAG_NAME}-focal
+    - run: ./utils/docker/publish_docker.sh stable
+      if: (github.event_name != 'workflow_dispatch' && !github.event.release.prerelease) || (github.event_name == 'workflow_dispatch' && github.event.inputs.is_release == 'true')
```
Comment (yury-s) [inline .github/workflows/publish_docker_release.yml:26]: Why not just check `if: (github.event.inputs.is_release == 'true')` and similarly inverted condition below ?

**Review by yury-s — APPROVED** [review]

File: utils/docker/build.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -e
+set +x
+
+if [[ ($1 == '--help') || ($1 == '-h') || ($1 == '') || ($2 == '') ]]; then
+  echo "usage: $(basename $0) {--arm64,--amd64} {bionic,focal} playwright:localbuild-focal"
```
Comment (yury-s) [inline utils/docker/build.sh:6]: let's remove bionic from here and other places

File: .github/workflows/publish_docker_release.yml (hunk 4 above)
Comment (mxschmitt) [reply .github/workflows/publish_docker_release.yml:26]: Lusha was working on it, I'd keep it for now.

File: utils/docker/publish_docker.sh (hunk 3 above)
Comment (mxschmitt) [reply utils/docker/publish_docker.sh:92]: Lets keep it to have a smaller diff in the future, its not used anyway (the bottom defines how the functions are called, and we call them only with focal)

**Pull request merged**